yt_dlp/output/console.py

```diff
@@ -0,0 +1,73 @@
+import sys
+
+from .hoodoo import BEL, CSI, TermCode
+from ..utils import supports_terminal_sequences, write_string
+
+
+class Console:
+    SAVE_TITLE = TermCode(f'{CSI}22;0t')
+    RESTORE_TITLE = TermCode(f'{CSI}23;0t')
+
+    def __init__(self, encoding=None, allow_title_change=False):
+        self.initialized = False
+        self.stream = None
+        self._stream_encoding = encoding
+
+        for stream in (sys.stderr, sys.stdout):
+            if supports_terminal_sequences(stream):
+                self.stream = stream
+                self.initialized = True
+
+        self.allow_title_change = allow_title_change
+        self._title_func = None
+
+        if not allow_title_change:
+            return
+
+        if self.initialized:
+            self._title_func = self._change_title_term_sequence
+            return
+
+        if sys.platform != 'win32':
+            return
+
+        import ctypes
+        if not hasattr(ctypes, 'windll'):
+            return
+
+        if not ctypes.windll.kernel32.GetConsoleWindow():
+            return
```

Comment:
This should no be needed, since we can safely call `SetConsoleWindow`.
The return value might be nonzero, but we can just ignore that.